src/main/java/fi/metatavu/dcfb/server/items/LocationController.java

```diff
@@ -0,0 +1,137 @@
+package fi.metatavu.dcfb.server.items;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import fi.metatavu.dcfb.server.persistence.dao.LocationDAO;
+import fi.metatavu.dcfb.server.persistence.model.Location;
+import fi.metatavu.dcfb.server.rest.model.LocationListSort;
+import fi.metatavu.dcfb.server.persistence.model.LocalizedEntry;
+import fi.metatavu.dcfb.server.search.handlers.LocationIndexHandler;
+import fi.metatavu.dcfb.server.search.searchers.LocationSearcher;
+import fi.metatavu.dcfb.server.search.searchers.SearchResult;
+
+@ApplicationScoped
+public class LocationController {
+
+  @Inject
+  private LocationSearcher locationSearcher;
+
+  @Inject
+  private LocationIndexHandler locationIndexHandler;
+  
+  @Inject
+  private LocationDAO locationDAO;
+
+  /**
+    * Creates new location
+    *
+    * @param slug slug
+    * @param additionalInformations additionalInformations
+    * @param country country
+    * @param latitude latitude
+    * @param longitude longitude
+    * @param name name
+    * @param postalCode postalCode
+    * @param postOffice postOffice
+    * @param streetAddress streetAddress
+    * @param createdAt createdAt
+    * @param modifiedAt modifiedAt
+    * @param lastModifier modifier
+    * @return created location
+    */
+  public Location createLocation(String slug, LocalizedEntry additionalInformations, String country, BigDecimal latitude, BigDecimal longitude, LocalizedEntry name, String postalCode, String postOffice, String streetAddress, UUID lastModifier) {
+    return locationDAO.create(UUID.randomUUID(), getUniqueSlug(slug), additionalInformations, country, latitude, longitude, name, postalCode, postOffice, streetAddress, lastModifier);
+  }
+
+  /**
+   * Finds an location
+   * 
+   * @param locationId location id
+   * @return location or null if not found
+   */
+  public Location findLocation(UUID locationId) {
+    return locationDAO.findById(locationId);
+  }
+  
+  /**
+   * Update location
+   *
+   * @param additionalInformations additionalInformations
+   * @param country country
+   * @param latitude latitude
+   * @param longitude longitude
+   * @param name name
+   * @param postalCode postalCode
+   * @param postOffice postOffice
+   * @param streetAddress streetAddress
+   * @param modifier modifier
+   * @return updated location
+   */
+  public Location updateLocation(Location location, LocalizedEntry additionalInformations, String country, BigDecimal latitude, BigDecimal longitude, LocalizedEntry name, String postalCode, String postOffice, String streetAddress, UUID modifier) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Method has 10 parameters, which is greater than 7 authorized. ([squid:S00107](https://sonarcloud.io/organizations/metatavu-oy/rules#rule_key=squid%3AS00107))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fi.metatavu.dcfb-api%3Adcfb-api&pullRequest=11&issues=AWQt_bhFYwZU9vvrs3Ko&open=AWQt_bhFYwZU9vvrs3Ko)

[ISSUE_KEY]: AWQt_bhFYwZU9vvrs3Ko